--- a/sam_3d_body/models/heads/mhr_head.py
+++ b/sam_3d_body/models/heads/mhr_head.py
@@ -103,7 +103,7 @@
         self.nonhand_param_idxs = nn.Parameter(
             torch.zeros(145).long(), requires_grad=False
         )
-
+        
         # Load MHR itself
         if MOMENTUM_ENABLED:
             self.mhr = MHR.from_files(
@@ -115,7 +115,7 @@
                 mhr_model_path,
                 map_location=("cuda" if torch.cuda.is_available() else "cpu"),
             )
-
+        # self.mhr.get_joint_names() dir(self.mhr)
         for param in self.mhr.parameters():
             param.requires_grad = False
 
--- a/sam_3d_body/models/meta_arch/sam3d_body.py
+++ b/sam_3d_body/models/meta_arch/sam3d_body.py
@@ -50,7 +50,7 @@
         self.register_buffer(
             "image_std", torch.tensor(self.cfg.MODEL.IMAGE_STD).view(-1, 1, 1), False
         )
-
+        
         # Create backbone feature extractor for human crops
         self.backbone = create_backbone(self.cfg.MODEL.BACKBONE.TYPE, self.cfg)
 
@@ -157,14 +157,14 @@
         self.hand_pe_layer = PositionEmbeddingRandom(self.backbone.embed_dims // 2)
 
         # Manually convert the torso of the model to fp16.
-        if self.cfg.TRAIN.USE_FP16:
-            self.convert_to_fp16()
-            if self.cfg.TRAIN.get("FP16_TYPE", "float16") == "float16":
-                self.backbone_dtype = torch.float16
-            else:
-                self.backbone_dtype = torch.bfloat16
-        else:
-            self.backbone_dtype = torch.float32
+        # if self.cfg.TRAIN.USE_FP16:
+        #     self.convert_to_fp16()
+        #     if self.cfg.TRAIN.get("FP16_TYPE", "float16") == "float16":
+        #         self.backbone_dtype = torch.float16
+        #     else:
+        #         self.backbone_dtype = torch.bfloat16
+        # else:
+        #     self.backbone_dtype = torch.float32
 
         self.ray_cond_emb = CameraEncoder(
             self.backbone.embed_dim,
@@ -1094,8 +1094,11 @@
             batch["ray_cond_hand"] = ray_cond[self.hand_batch_idx].clone()
         ray_cond = None
 
+        # image_embeddings = self.backbone(
+        #     x.type(self.backbone_dtype), extra_embed=ray_cond
+        # )  # (B, C, H, W)
         image_embeddings = self.backbone(
-            x.type(self.backbone_dtype), extra_embed=ray_cond
+            x, extra_embed=ray_cond
         )  # (B, C, H, W)
 
         if isinstance(image_embeddings, tuple):
--- a/sam_3d_body/sam_3d_body_estimator.py
+++ b/sam_3d_body/sam_3d_body_estimator.py
@@ -111,7 +111,7 @@
             if image_format == "rgb":
                 img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
                 image_format = "bgr"
-            print("Running object detector...")
+            # print("Running object detector...")
             boxes = self.detector.run_human_detection(
                 img,
                 det_cat_id=det_cat_id,
@@ -119,7 +119,7 @@
                 nms_thr=nms_thr,
                 default_to_full_image=False,
             )
-            print("Found boxes:", boxes)
+            # print("Found boxes:", boxes)
             self.is_crop = True
         else:
             boxes = np.array([0, 0, width, height]).reshape(1, 4)
@@ -167,7 +167,7 @@
             cam_int = cam_int.to(batch["img"])
             batch["cam_int"] = cam_int.clone()
         elif self.fov_estimator is not None:
-            print("Running FOV estimator ...")
+            # print("Running FOV estimator ...")
             input_image = batch["img_ori"][0].data
             cam_int = self.fov_estimator.get_cam_intrinsics(input_image).to(
                 batch["img"]
@@ -211,7 +211,6 @@
                     "mask": masks[idx] if masks is not None else None,
                     "pred_joint_coords": out["pred_joint_coords"][idx],
                     "pred_global_rots": out["joint_global_rots"][idx],
-                    "mhr_model_params": out["mhr_model_params"][idx],
                 }
             )
 
--- a/sam_3d_body/utils/checkpoint.py
+++ b/sam_3d_body/utils/checkpoint.py
@@ -109,6 +109,7 @@
             f'missing keys in source state_dict: {", ".join(missing_keys)}\n'
         )
 
+
     if len(err_msg) > 0:
         err_msg.insert(0, "The model and loaded state dict do not match exactly\n")
         err_msg = "\n".join(err_msg)
--- a/sam_3d_body/visualization/renderer.py
+++ b/sam_3d_body/visualization/renderer.py
@@ -271,13 +271,12 @@
         #     baseColorFactor=(*mesh_base_color, 1.0))
         vertex_colors = np.array([(*mesh_base_color, 1.0)] * vertices.shape[0])
         mesh = trimesh.Trimesh(
-            vertices.copy() + camera_translation,
+            vertices.copy() + camera_translation,  # Body and camera axes agree, so only translation is needed.
             self.faces.copy(),
             vertex_colors=vertex_colors,
         )
 
         # mesh = trimesh.Trimesh(vertices.copy(), self.faces.copy())
-
         rot = trimesh.transformations.rotation_matrix(np.radians(rot_angle), rot_axis)
         mesh.apply_transform(rot)
 
--- a/tools/build_detector.py
+++ b/tools/build_detector.py
@@ -5,7 +5,6 @@
 
 import numpy as np
 import torch
-from PIL import Image
 
 
 class HumanDetector:
@@ -19,47 +18,8 @@
 
             self.detector = self.detector.to(self.device)
             self.detector.eval()
-        elif name == "sam3":
-            from sam3.model_builder import build_sam3_image_model
-            from sam3.model.sam3_image_processor import Sam3Processor
-            
-            self.detector = build_sam3_image_model()
-            self.processor = Sam3Processor(self.detector)
-            self.detector_func = lambda detector, img, **kwargs: self.sam3_run(
-                img, **kwargs
-            )
         else:
             raise NotImplementedError
-        
-    def sam3_run(self, img, det_cat_id: int = 0, bbox_thr: float = 0.5, **kwargs):
-        # switch bgr to rgb 
-        img = img[:, :, ::-1].copy()
-        img = Image.fromarray(img.astype('uint8'), 'RGB')
-        inference_state = self.processor.set_image(img)
-        # Prompt the model with text
-        output = self.processor.set_text_prompt(state=inference_state, prompt="person")
-
-        # Get the masks, bounding boxes, and scores
-        masks, boxes, scores = output["masks"], output["boxes"], output["scores"]
-        
-        confident_idx = scores > bbox_thr
-        boxes = boxes[confident_idx].cpu().numpy()
-        
-        # resize the box with a scale factor 1.2
-        scale = 1.2
-        enlarged_boxes = []
-        for box in boxes:
-            x1, y1, x2, y2 = box
-            cx = (x1 + x2) / 2
-            cy = (y1 + y2) / 2
-            w = (x2 - x1) * scale
-            h = (y2 - y1) * scale
-            new_x1 = max(cx - w / 2, 0)
-            new_y1 = max(cy - h / 2, 0)
-            new_x2 = cx + w / 2
-            new_y2 = cy + h / 2
-            enlarged_boxes.append([new_x1, new_y1, new_x2, new_y2])
-        return np.array(enlarged_boxes)
 
     def run_human_detection(self, img, **kwargs):
         return self.detector_func(self.detector, img, **kwargs)
@@ -82,11 +42,12 @@
         )
 
     detectron2_cfg = LazyConfig.load(str(cfg_path))
-    detectron2_cfg.train.init_checkpoint = (
-        "https://dl.fbaipublicfiles.com/detectron2/ViTDet/COCO/cascade_mask_rcnn_vitdet_h/f328730692/model_final_f05665.pkl"
-        if path == ""
-        else os.path.join(path, "model_final_f05665.pkl")
-    )
+    # detectron2_cfg.train.init_checkpoint = (
+    #     "https://dl.fbaipublicfiles.com/detectron2/ViTDet/COCO/cascade_mask_rcnn_vitdet_h/f328730692/model_final_f05665.pkl"
+    #     if path == ""
+    #     else os.path.join(path, "model_final_f05665.pkl")
+    # )
+    detectron2_cfg.train.init_checkpoint = "model/ViTDet/model_final_f05665.pkl"
     for i in range(3):
         detectron2_cfg.model.roi_heads.box_predictors[i].test_score_thresh = 0.25
     detector = instantiate(detectron2_cfg.model)
--- a/tools/build_fov_estimator.py
+++ b/tools/build_fov_estimator.py
@@ -25,6 +25,7 @@
 
     if path == "":
         path = "Ruicheng/moge-2-vitl-normal"
+    path = "model/moge-2-vitl-normal/model.pt"
     moge_model = MoGeModel.from_pretrained(path).to(device)
     return moge_model
 
--- a/tools/build_sam.py
+++ b/tools/build_sam.py
@@ -2,7 +2,6 @@
 
 import torch
 import numpy as np
-from PIL import Image
 
 
 class HumanSegmentor:
@@ -13,10 +12,7 @@
             print("########### Using human segmentor: SAM2...")
             self.sam = load_sam2(device, **kwargs)
             self.sam_func = run_sam2
-        elif name == "sam3":
-            print("########### Using human segmentor: SAM3...")
-            self.sam = load_sam3(device, **kwargs)
-            self.sam_func = run_sam3
+
         else:
             raise NotImplementedError
     
@@ -25,8 +21,10 @@
         
 
 def load_sam2(device, path):
-    checkpoint = f"{path}/checkpoints/sam2.1_hiera_large.pt"
-    model_cfg = "configs/sam2.1/sam2.1_hiera_l.yaml"
+    checkpoint = "model/sam2.1-hiera-large/sam2.1_hiera_large.pt"
+    model_cfg = "model/sam2.1-hiera-large/sam2.1_hiera_l.yaml"
+    # checkpoint = f"{path}/checkpoints/sam2.1_hiera_large.pt"
+    # model_cfg = "configs/sam2.1/sam2.1_hiera_l.yaml"
 
     import sys
     sys.path.append(path)
@@ -36,15 +34,6 @@
     predictor = SAM2ImagePredictor(build_sam2(model_cfg, checkpoint, device=device))
     predictor.model.eval()
 
-    return predictor
-
-
-def load_sam3(device, path):
-    from sam3.model_builder import build_sam3_image_model
-    from sam3.model.sam3_image_processor import Sam3Processor
-    
-    model = build_sam3_image_model()
-    predictor = Sam3Processor(model)
     return predictor
 
 
@@ -73,23 +62,6 @@
             # cv2.imwrite(os.path.join(save_dir, f"{os.path.basename(image_path)[:-4]}_mask_{i}.jpg"), (mask_1 * 255).astype(np.uint8))
         all_masks = np.stack(all_masks)
         all_scores = np.stack(all_scores)
-
+    
     return all_masks, all_scores
-
-
-def run_sam3(sam_predictor, img, boxes):
-    # switch bgr to rgb 
-    img = img[:, :, ::-1].copy()
-    img = Image.fromarray(img.astype('uint8'), 'RGB')
-    inference_state = sam_predictor.set_image(img)
-    # Prompt the model with text
-    output = sam_predictor.set_text_prompt(state=inference_state, prompt="person")
-
-    # Get the masks, bounding boxes, and scores
-    masks, boxes, scores = output["masks"], output["boxes"], output["scores"]
-    score_threshold = 0.5
-    confident_idx = scores > score_threshold
-    masks = masks[confident_idx].float().squeeze(1).cpu().numpy()
-    scores = scores[confident_idx].cpu().numpy()
-
-    return masks, scores
\ No newline at end of file
+        
\ No newline at end of file
